File: leetcode/1768.merge-strings-alternately.go
/*
 * @lc app=leetcode id=1768 lang=golang
 *
 * [1768] Merge Strings Alternately
 */

// @lc code=start
func mergeAlternately(word1 string, word2 string) string {
	var len1, len2 int = len(word1), len(word2)
    var result string
	for i := 0; i < len1 || i < len2; i++ {
		if i < len1 {
			result += string(word1[i])
		}
		if i < len2 {
			result += string(word2[i])
		}
	}
	return result
    
}
// @lc code=end

File: leetcode/1768.merge-strings-alternately.kt
/*
 * @lc app=leetcode id=1768 lang=kotlin
 *
 * [1768] Merge Strings Alternately
 */

// @lc code=start
class Solution {
    fun mergeAlternately(word1: String, word2: String): String {
        val len1 = word1.length
        val len2 = word2.length
        val result = StringBuilder()
        (0..< maxOf(len1, len2)).forEach {
            if (it < len1) {
                result.append(word1[it])
            }
            if (it < len2) {
                result.append(word2[it])
            }
        }
        return result.toString()
    }
}
// @lc code=end

